### ISSUE TITLE
False positive for DirectInvocationOnMock

### ISSUE THREAD

**Issue by moonfruit** [google/error-prone#3396]
Code like this should be allowed for `DirectInvocationOnMock`:

```java
import static org.mockito.Mockito.mock;
import static org.mockito.BDDMockito.given;

@Test
void test() {
    Caller caller = mock(Caller.class);
    given(caller.call()).willReturn(1);
}
```

**Comment by RobertTheBlair** [comment]
Thanks for creating this issue.  I was going to make one.  Is this still under consideration?

**Comment by RobertTheBlair** [comment]
https://github.com/google/error-prone/pull/3731 will address this issue
Hope it's not too late @moonfruit

**Comment by RobertTheBlair** [comment]
Still looking to get this reviewed & approved - everything should be ready, except the build does not pass as I've not been approved by a maintainer to run workflows.
